src/main/java/it/pagopa/pn/deliverypush/webhook/WebhookClientCertImpl.java

```diff
@@ -0,0 +1,131 @@
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.client.HttpClient;
+import org.apache.http.conn.ssl.TrustStrategy;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import javax.net.ssl.SSLContext;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.security.*;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Base64;
+import java.util.List;
+
+@Slf4j
+@Component( value = "WebhookClientCertImpl")
+public class WebhookClientCertImpl implements WebhookClient{
+
+    private static final char[] KEYSTORE_PASSWORD = "password".toCharArray();
+
+    private RestTemplate restTemplate;
+
+
+    @Autowired
+    private CertCfg certCfg;
+
+    WebhookClientCertImpl(CertCfg certCfg ) {
+        this.certCfg = certCfg;
+    }
+
+    private RestTemplate buildRestTemplate() {
+        RestTemplateBuilder builder = new RestTemplateBuilder();
+
+
+        SSLContext sslContext = null;
+        try {
+            String stringCert = certCfg.getClientCertificatePem();
+            String clientKeyPem = certCfg.getClientKeyPem();
+
+            if ( stringCert != null && !stringCert.isEmpty() && clientKeyPem != null && !clientKeyPem.isEmpty() ) {
+                X509Certificate clientCert = buildX509CertificateFromPemString(stringCert);
+
+                KeyStore keyStore = newKeyStore();
+
+                keyStore.setCertificateEntry( "client_cert", clientCert );
+                keyStore.setKeyEntry("client_key", buildKeyFromPemString(clientKeyPem),
+                        KEYSTORE_PASSWORD,
+                        new Certificate[] {clientCert});
+
+                TrustStrategy acceptingTrustStrategy = (X509Certificate[] chain, String authType) -> true;
+                sslContext = SSLContextBuilder.create()
+                        .loadTrustMaterial( null, acceptingTrustStrategy )
+                        .loadKeyMaterial(keyStore, KEYSTORE_PASSWORD)
+                        .build();
+            } else {
+                log.error( "Unable to retrieve cert/key from env vars" );
+                throw new PnInternalException( "Unable to retrieve cert/key from env vars" );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Exception [PnInternalException] is explicitly thrown on this line but is caught in the outer catch clause with generic exception on line [79].  Consider adding a catch clause of this specific exception in the outer try statement.